image-processing/main.py

```diff
@@ -5,10 +5,48 @@
 # Then runs the provided R-script to generate processed images
 
 import os
+import cv2
+import requests
+import numpy as np
+
 from google.cloud import vision
 from google.cloud import storage
 
 
+def face_check(face):
+
+    EMOTION_THRESHOLD = 3
+    LIGHTING_THRESHOLD = 3
+    BLURRY_THRESHOLD = 2
+    ANGLE_THRESHOLD = 10.0
+
+    if (
+        face.joy_likelihood > EMOTION_THRESHOLD
+        or face.anger_likelihood > EMOTION_THRESHOLD
+        or face.surprise_likelihood > EMOTION_THRESHOLD
+        or face.sorrow_likelihood > EMOTION_THRESHOLD
+    ):
+        return "Please make sure your image has neutral facial expression."
+
+    if face.under_exposed_likelihood > LIGHTING_THRESHOLD:
+        return "Please make sure your image is well-lit."
+
+    if face.blurred_likelihood > BLURRY_THRESHOLD:
+        return "Please make sure your image is clear."
+
+    if (
+        face.roll_angle < -ANGLE_THRESHOLD
+        or face.roll_angle > ANGLE_THRESHOLD
+        or face.pan_angle < -ANGLE_THRESHOLD
+        or face.pan_angle > ANGLE_THRESHOLD
+        or face.tilt_angle < -ANGLE_THRESHOLD
+        or face.tilt_angle > ANGLE_THRESHOLD
+    ):
+        return "Please face straight towards your camera, avoid tiliting."
+
+    return "Success"
```

Comment:
hmm I'd return sth else rather than a string "success". Maybe a bool would suffice, then you could raise exception with specific messages? What do you think?

Reply:
Good idea! I added a custom exception